drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2021 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_uart
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/uart.h>
+#include <kernel.h>
+#include <pm/device.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(uart_ite_it8xxx2, LOG_LEVEL_ERR);
+
+#define DRV_CONFIG(dev) \
+	((const struct uart_it8xxx2_config *)(dev)->config)
+
+struct uart_it8xxx2_config {
+	uint8_t port;
+	/* GPIO handle */
+	const struct device *gpio_dev;
+	/* UART handle */
+	const struct device *uart_dev;
+};
+
+enum uart_port_num {
+	UART1 = 1,
+	UART2,
+};
+
+#ifdef CONFIG_PM_DEVICE
+__weak void uart1_wui_isr_late(void) { }
+__weak void uart2_wui_isr_late(void) { }
+
+void uart1_wui_isr(const struct device *gpio, struct gpio_callback *cb,
+		   uint32_t pins)
+{
+	/* Disable interrupts on UART1 RX pin to avoid repeated interrupts. */
+	gpio_pin_interrupt_configure(gpio, (find_msb_set(pins) - 1),
+				     GPIO_INT_DISABLE);
+
+	if (uart1_wui_isr_late) {
+		uart1_wui_isr_late();
+	}
+}
+
+void uart2_wui_isr(const struct device *gpio, struct gpio_callback *cb,
+		   uint32_t pins)
+{
+	/* Disable interrupts on UART2 RX pin to avoid repeated interrupts. */
+	gpio_pin_interrupt_configure(gpio, (find_msb_set(pins) - 1),
+				     GPIO_INT_DISABLE);
+
+	if (uart2_wui_isr_late) {
+		uart2_wui_isr_late();
+	}
+}
+
+static inline int uart_it8xxx2_pm_action_cb(const struct device *dev,
+					    enum pm_device_action action)
+{
+	const struct uart_it8xxx2_config *config = DRV_CONFIG(dev);
+	int ret = 0;
+
+	switch (action) {
+	/* Next device power state is in active. */
+	case PM_DEVICE_ACTION_RESUME:
+		/* To do when EC is resuming. */
+		break;
+	/* Next device power state is deep doze mode */
+	case PM_DEVICE_ACTION_SUSPEND:
+		if (config->port == UART1) {
+			/* Enable UART1 WUI */
+			ret = gpio_pin_interrupt_configure(config->gpio_dev, 0,
+							   GPIO_INT_TRIG_LOW);
+		} else if (config->port == UART2) {
+			/* Enable UART2 WUI */
+			ret = gpio_pin_interrupt_configure(config->gpio_dev, 1,
+							   GPIO_INT_TRIG_LOW);
+		}
```

Comment:
I think you should add the GPIO pin number as a device tree property.

You can the eliminate this if branch.

Reply:
Done.